meson_options.txt

```diff
@@ -0,0 +1,12 @@
+option('probe', type: 'combo', choices : ['96b_carbon', 'blackpill-f401cc', 'blackpill-f401ce', 'blackpill-f411ce', 'bluepill', 'f072', 'f3', 'f4discovery', 'hydrabus', 'launchpad-icdi', 'native', 'stlink', 'stlinkv3', 'swlink'], value: 'native', description: 'Probe host')
```

Comment:
Noted, I will try to revert that.

This particular reorganization has lost it's purpose amidst all the refactoring/iterations I have done. By now I don't remember the details, but the buildsystem extracted the available probe platforms from the directories in `probe/` every dir was a distinct probe, sharing the sir with the common platform code would mean `common/` and  `stm32/` would be considered probes, for example.
This is no longer relevant though, the way it's written like now allows the directory to be shared, though I still would like to see a reorganization of the like later

Reply:
Agreed that reorganisation is required - including the merge-down of the 3 blackpill platforms; Think we'll coordinate that with you in a follow-up PR once the Makefile build system is KConfig-ised so it can be done in a clean step. Doing the overhaul of the Makefile build system is the next thing on our TODO list after we get this PR merged.